arangod/Replication2/Storage/WAL/WalManager.cpp

```diff
@@ -0,0 +1,50 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Manuel Pöter
+////////////////////////////////////////////////////////////////////////////////
+
+#include "WalManager.h"
+#include <filesystem>
+
+#include "Basics/Exceptions.h"
+#include "Basics/FileUtils.h"
+#include "Basics/Result.h"
+#include "Replication2/Storage/WAL/FileManager.h"
+#include "Replication2/Storage/WAL/IFileManager.h"
+
+namespace arangodb::replication2::storage::wal {
+
+WalManager::WalManager(std::string folderPath)
+    : _folderPath(std::move(folderPath)) {
+  std::filesystem::create_directories(_folderPath);
```

Comment:
As with my other comment regarding sync and directory entries, I think we have to open directories along the path and call `fsync` on their fds here, so the directories will actually be visible. E.g. if `/a/b/` already exists, and we create `/a/b/c/d/`, we'll have to open&fsync at least `b` and `c`, assuming `a` was already synced after `b` was created.